view/src/client.rs

```diff
@@ -705,7 +705,14 @@ where
                     tx_hash: tx_hash.as_bytes().to_vec(),
                 }),
             );
-            rsp.await?.into_inner().txp.unwrap().try_into()
+            // If no txp was supplied, use the default (empty) perspective.
+            Ok(rsp
+                .await?
+                .into_inner()
+                .txp
+                .map(|txp| txp.try_into())
+                .transpose()?
+                .unwrap_or_default())
```

Comment:
I don't love this code, it feels really non-intuitive to read and understand what's going on. Probably it should be rewritten to not use so many combinators, or rather, it should be rewritten so that the option/result combinators are doing the non-semantically meaningful "data juggling", and have explicit control flow for the semantically meaningful "if no txp supplied, return default" behavior